### PR TITLE
hdf5 and dependents: downgrade and use `libaec`

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Related to #96930

### CODE REVIEW

**Comment by cho-m** [comment]
This will probably need revision bumps based on #94244. However, may wait for some other PRs to get merged to avoid conflicts.

**Comment by cho-m** [comment]
Seems like `vtk` doesn't like new clang version 13.1.6 / XCode 13.3 as it segfaulted.

File: Formula/vtk.rb [hunk 1]
```diff
@@ -46,16 +46,30 @@ class Vtk < Formula
...
+    depends_on "libaec"
     depends_on "mesa-glu"
   end
 
   fails_with gcc: "5"
 
+  # clang: error: unable to execute command: Segmentation fault: 11
+  # clang: error: clang frontend command failed due to signal (use -v to see invocation)
+  # Apple clang version 13.1.6 (clang-1316.0.21.2)
+  fails_with :clang if DevelopmentTools.clang_build_version == 1316 && Hardware::CPU.arm?
```
Comment (cho-m) [inline Formula/vtk.rb:64]: Probably going to segfault on Intel mac too, but testing ARM-only just in case.

**Review by SMillerDev — DISMISSED** [review]

**Comment by cho-m** [comment]
I'll merge this now to hopefully avoid any merge conflicts and get on to removing `szip`.

Will follow up with `csound` upstream for fixing Python.h location. Since it only failed to build on Monterey, it is probably related to Python 2 removal.

**Comment by cho-m** [comment]
Sadly, will need to figure out how to deal with downgrade since previous version is already used.

Not sure if we can `revision 1` this without causing an audit issue.

**Comment by SMillerDev** [comment]
You either need a version schema change or revision bump.

**Comment by cho-m** [comment]
There is a `version_scheme` which avoids the audit error for downgrade.

The problem seems to be that the uploaded packages don't use that in naming, so it clashes with existing https://github.com/Homebrew/homebrew-core/pkgs/container/core%2Fhdf5/10917206?tag=1.12.1

So, would probably need both `version_scheme 1` + `revision 1` to bypass audit for downgrade and avoid conflict with the existing package.

---

Probably will hit merge conflict with #97987, so will wait on that PR to be merged and then rebase before next run.

**Comment by SMillerDev** [comment]
It should overwrite if it's a version scheme change

**Comment by cho-m** [comment]
> It should overwrite if it's a version scheme change

Not familiar with upload behavior, but error message is:
```
/usr/bin/skopeo inspect --raw docker://ghcr.io/homebrew/core/hdf5:1.12.1 --creds=brewtestbot:******
Error: docker://ghcr.io/homebrew/core/hdf5:1.12.1 already exists!
Error: Kernel.exit
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:160:in `exit'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:160:in `odie'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:234:in `upload_bottle'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:62:in `block in upload_bottles'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:61:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/github_packages.rb:61:in `upload_bottles'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/pr-upload.rb:163:in `pr_upload'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:110:in `<main>'
```

File: Formula/hdf5.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
 class Hdf5 < Formula
   desc "File format designed to store large amounts of data"
   homepage "https://www.hdfgroup.org/HDF5"
-  url "https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.13/hdf5-1.13.0/src/hdf5-1.13.0.tar.bz2"
-  sha256 "1826e198df8dac679f0d3dc703aba02af4c614fd6b7ec936cf4a55e6aa0646ec"
+  url "https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.12/hdf5-1.12.1/src/hdf5-1.12.1.tar.bz2"
+  sha256 "aaf9f532b3eda83d3d3adc9f8b40a9b763152218fa45349c3bc77502ca1f8f1c"
   license "BSD-3-Clause"
+  revision 1
+  version_scheme 1
```
Comment (cho-m) [inline Formula/hdf5.rb:8]: This time trying both `revision` + `version_scheme` to bypass upload issue.

**Comment by cho-m** [comment]
I think the previous merge attempt uploaded unwanted `caffe` bottle. The formula is still on revision 36 but 37 exists https://github.com/Homebrew/homebrew-core/pkgs/container/core%2Fcaffe/17788938?tag=1.0_37
```
/usr/bin/skopeo inspect --raw docker://ghcr.io/homebrew/core/caffe:1.0_37 --creds=brewtestbot:******
Error: docker://ghcr.io/homebrew/core/caffe:1.0_37 already exists!
Error: Kernel.exit
```

Might need to see if we can delete that upload

**Comment by cho-m** [comment]
Tried deleting `caffe` and it got past previous error, but failed on `alpscore`.

I just went through and tried to remove most of incorrectly uploaded bottled; however, I can't remove incorrectly uploaded  `alpscore` as it looks like write permissions are different on that GitHub package https://github.com/Homebrew/homebrew-core/pkgs/container/core%2Falpscore

Could be fact that it is newer or the way it was uploaded. Worst case, would need to try extra revision bump

**Comment by SMillerDev** [comment]
I'm removing it now

**Pull request closed without merging**